numpy/core/tests/test_scalarbuffer.py

```diff
@@ -78,8 +78,29 @@ def test_void_scalar_structured_data(self):
         assert_equal(mv_x.itemsize, mv_a.itemsize)
         assert_equal(mv_x.format, mv_a.format)
 
-    def test_invalid_buffer_format(self):
-        # datetime64 cannot be used in a buffer yet
+    def test_datetime_memoryview(self):
+        # gh-11656
+        # Values verified with v1.13.3, shape is not () as in test_scalar_dim
+        def as_dict(m):
+            return dict(strides=m.strides, shape=m.shape, itemsize=m.itemsize,
+                        ndim=m.ndim, format=m.format)
+
+        dt1 = np.datetime64('2016-01-01')
+        dt2 = np.datetime64('2017-01-01')
+        expected = {'strides': (1,), 'itemsize': 1, 'ndim': 1,
+                    'shape': (8,), 'format': 'B'}
+        v = memoryview(dt1)
+        res = as_dict(v) 
+        assert_equal(res, expected)
+
+        v = memoryview(dt2 - dt1)
+        res = as_dict(v)
+        assert_equal(res, expected)
+
         dt = np.dtype([('a', int), ('b', 'M8[s]')])
         a = np.empty(1, dt)
-        assert_raises((ValueError, BufferError), memoryview, a[0])
+        m = memoryview(a[0])
+        assert_equal(m.format, 'T{l:a:B:b:}')
+        assert_equal(m.itemsize, 16)
```

Comment:
Taken together, these two lines violate PEP3118. If we can't produce a valid format string, we should just error.

Reply:
Using `'uint16'` instead of `int` in the dtype would make the `itemsize=16` valid, correct?

Reply:
ahh, no sorry, I see the problem.